src/jit.rs

```diff
@@ -1114,8 +1114,11 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
 
         match dst {
             Value::Register(reg) => {
-                // Move guest_target_address into REGISTER_MAP[FRAME_PTR_REG]
-                self.emit_ins(X86Instruction::mov(OperandSize::S64, reg, REGISTER_MAP[FRAME_PTR_REG]));
+                // REGISTER_SCRATCH contains self.pc, and we must store it for proper error handling.
+                // We can discard the value if callx succeeds, so we are not incrementing the stack pointer (RSP).
+                self.emit_ins(X86Instruction::store(OperandSize::S64, REGISTER_SCRATCH, RSP, X86IndirectAccess::OffsetIndexShift(-24, RSP, 0)));
```

Comment:
Would it be possible to use another x86 register here? I'm concerned that adding loads and stores may impact performance.

Reply:
It is the same store that was inside the anchor before, just inlined here. The only thing this makes worse is more bytes emitted per `callx`.